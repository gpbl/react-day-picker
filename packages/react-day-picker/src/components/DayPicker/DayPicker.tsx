import { isAfter, isBefore, startOfMonth } from 'date-fns';
import * as React from 'react';

import { Months } from '../Months';
import { DEFAULT_PROPS } from './defaults/props';

/**
 * Render a date picker component.
 *
 * @category Component
 */
export function DayPicker(props = DEFAULT_PROPS): JSX.Element {
  const isControlled = Boolean(props.month);

  let initialMonth = startOfMonth(
    props.initialMonth || (props.today !== 'off' && props.today) || new Date()
  );

  if (props.toMonth && isAfter(initialMonth, startOfMonth(props.toMonth))) {
    initialMonth = startOfMonth(props.toMonth);
  }

  if (
    props.fromMonth &&
    isBefore(initialMonth, startOfMonth(props.fromMonth))
  ) {
    initialMonth = startOfMonth(props.fromMonth);
  }

  const [currentMonth, setCurrentMonth] = React.useState(
    startOfMonth(initialMonth)
  );

  function handleMonthChange(month: Date, e: React.MouseEvent): void {
    setCurrentMonth(month);
    props.onMonthChange?.(month, e);
  }
  return (
    <Months
      {...props}
      onMonthChange={!isControlled ? handleMonthChange : props.onMonthChange}
      month={isControlled ? props.month : currentMonth}
    />
  );
}
