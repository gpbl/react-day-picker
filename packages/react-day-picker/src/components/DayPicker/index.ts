export * from './DayPicker';
export * from './types';
