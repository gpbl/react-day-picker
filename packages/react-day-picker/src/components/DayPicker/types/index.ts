export * from './CustomComponents';
export * from './DayClickEventHandler';
export * from './DayPickerClassNames';
export * from './DayPickerProps';
export * from './DayPickerStyles';
export * from './DefaultModifier';
export * from './Matcher';
export * from './Modifiers';
export * from './ModifiersMatchers';
export * from './MonthChangeEventHandler';
export * from './UIElement';
