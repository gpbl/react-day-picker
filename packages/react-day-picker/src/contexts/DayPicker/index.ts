export * from './useDayPicker';
export * from './DayPickerContext';
