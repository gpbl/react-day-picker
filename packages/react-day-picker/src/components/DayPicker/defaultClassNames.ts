import { DayPickerClassNames } from 'types';

const classNames: DayPickerClassNames = {
  /** Root element */
  root: 'rdp',
  caption: 'rdp-caption',

  // Day Component
  day: 'rdp-day',
  dayWrapper: 'rdp-day_wrapper',

  // Month Component
  month: 'rdp-month',
  monthTable: 'rdp-table',
  monthTbody: 'rdp-body',
  months: 'rdp-months',

  // Head Components
  head: 'rdp-head',
  headRow: 'rdp-head-row',
  headWeekNumber: 'rdp-head_weeknumber',
  headWeekName: 'rdp-head-cell',

  // Navigation Component
  nav: 'rdp-nav',
  navPrev: 'rdp-nav_prev',
  navNext: 'rdp-nav_next',

  // Week Component
  week: 'rdp-row',
  weekDay: 'rdp-cell',
  weekWeeknumber: 'rdp-row_head',

  // WeekNumber component
  weekNumber: 'rdp-rownumber',

  // Modifiers
  selected: 'rdp-day_selected',
  disabled: 'rdp-day_disabled',
  today: 'rdp-day_today',
  aftermonth: 'rdp-day_outside',
  beforemonth: 'rdp-day_outside'
};

export default classNames;
