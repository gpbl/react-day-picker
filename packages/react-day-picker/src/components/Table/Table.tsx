import React from 'react';

import { useDayPicker } from 'contexts/DayPicker';

import { getMonthWeeks } from './utils/getMonthWeeks';

/** The props for the {@link Table} component. */
export interface TableProps {
  /** The ID of the label of the table (the same given to the Caption). */
  ['aria-labelledby']?: string;
  /** The month where the table is displayed. */
  displayMonth: Date;
}

/** Render the table with the calendar. */
export function Table(props: TableProps): JSX.Element {
  const {
    locale,
    classNames,
    styles,
    hideHead,
    fixedWeeks,
    components: { Head, Row, Footer },
    weekStartsOn,
    firstWeekContainsDate,
    ISOWeek
  } = useDayPicker();

  const weeks = getMonthWeeks(props.displayMonth, {
    useFixedWeeks: Boolean(fixedWeeks),
    ISOWeek,
    locale,
    weekStartsOn,
    firstWeekContainsDate
  });

  return (
    <table
      className={classNames.table}
      style={styles.table}
      role="grid"
      aria-labelledby={props['aria-labelledby']}
    >
      {!hideHead && <Head />}
      <tbody className={classNames.tbody} style={styles.tbody}>
        {weeks.map((week) => (
          <Row
            displayMonth={props.displayMonth}
            key={week.weekNumber}
            dates={week.dates}
            weekNumber={week.weekNumber}
          />
        ))}
      </tbody>
      <Footer />
    </table>
  );
}
