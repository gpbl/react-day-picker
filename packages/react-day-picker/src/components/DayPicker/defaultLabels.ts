import { format } from 'date-fns';
import { LabelsFormatters } from '../../types/LabelsFormatters';

import { defaultProps } from './defaultProps';

export const defaultLabels: LabelsFormatters = {
  navNext: (month, props) => {
    const formatCaption = props?.formatCaption ?? defaultProps.formatCaption;
    return `Go to next month: ${formatCaption(month, props)}`;
  },
  navPrev: (month, props) => {
    const formatCaption = props?.formatCaption ?? defaultProps.formatCaption;
    return `Go to previous month: ${formatCaption(month, props)}`;
  },
  day: (day, _, props) => {
    return format(day, 'PPPP', { locale: props.locale });
  },
  weekday: (day, props) => {
    return props.formatWeekdayName(day, { locale: props.locale }, 'cccc');
  },
  weekNumber: (n, props) => {
    return `Week n. ${n}`;
  }
};
