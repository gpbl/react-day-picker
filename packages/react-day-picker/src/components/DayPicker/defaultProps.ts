import english from 'date-fns/locale/en-US';
import { DayPickerProps } from 'types';

import { Caption } from '../Caption/Caption';
import { Day } from '../Day/Day';
import { Navigation } from '../Navigation';
import { NextIcon } from '../NextIcon';
import { PrevIcon } from '../PrevIcon';
import { Week } from '../Week';
import classNames from './defaultClassNames';
import { defaultLabels } from './defaultLabels';
import { formatCaption } from './formatters/formatCaption';
import { formatDay } from './formatters/formatDay';
import { formatWeekdayName } from './formatters/formatWeekdayName';
import { formatWeekNumber } from './formatters/formatWeekNumber';

export const defaultProps: DayPickerProps = {
  labelsFormatters: defaultLabels,
  enableOutsideDaysClick: false,
  classNames,
  className: '',
  components: {
    Caption,
    Day,
    Navigation,
    NextIcon,
    PrevIcon,
    Week
  },
  fixedWeeks: false,
  formatCaption,
  formatDay,
  formatWeekdayName,
  formatWeekNumber,
  initialMonth: new Date(),
  locale: english,
  modifiersClassNames: {},
  month: new Date(),
  numberOfMonths: 1,
  pagedNavigation: false,
  reverseMonths: false,
  showCaption: true,
  showHead: true,
  showNavigation: true,
  showOutsideDays: false,
  showWeekNumber: false,
  today: new Date()
};
