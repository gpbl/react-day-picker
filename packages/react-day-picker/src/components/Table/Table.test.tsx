import React from 'react';

import { DayPickerProps } from 'DayPicker';

import { customRender } from 'test/render/customRender';

import { Table, TableProps } from './Table';

function setup(props: TableProps, dayPickerProps?: DayPickerProps) {
  return customRender(<Table {...props} />, dayPickerProps);
}

const props: TableProps = {
  displayMonth: new Date(2020, 1)
};

test('should render correctly', () => {
  const { container } = setup(props);
  expect(container.firstChild).toMatchSnapshot();
});

describe('when showing the week numbers', () => {
  const dayPickerProps = { showWeekNumber: true };
  test('should render correctly', () => {
    const { container } = setup(props, dayPickerProps);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('when using custom components', () => {
  const dayPickerProps: DayPickerProps = {
    components: {
      Head: () => <div>CustomHead </div>,
      Row: () => <div>CustomRow </div>,
      Footer: () => <div>CustomFooter</div>
    }
  };
  test('should render correctly', () => {
    const { container } = setup(props, dayPickerProps);
    expect(container.firstChild).toMatchSnapshot();
  });
});
