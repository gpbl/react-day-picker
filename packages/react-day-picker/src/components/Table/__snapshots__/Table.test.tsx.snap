// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<table
  class="rdp-table"
  role="grid"
>
  <thead
    class="rdp-head"
  >
    <tr
      class="rdp-head_row"
    >
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Su
        </span>
        <span
          class="rdp-vhidden"
        >
          Sunday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Mo
        </span>
        <span
          class="rdp-vhidden"
        >
          Monday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Tu
        </span>
        <span
          class="rdp-vhidden"
        >
          Tuesday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          We
        </span>
        <span
          class="rdp-vhidden"
        >
          Wednesday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Th
        </span>
        <span
          class="rdp-vhidden"
        >
          Thursday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Fr
        </span>
        <span
          class="rdp-vhidden"
        >
          Friday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Sa
        </span>
        <span
          class="rdp-vhidden"
        >
          Saturday
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    class="rdp-tbody"
  >
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      >
        <div
          aria-label="1st February (Saturday)"
          class="rdp-day"
        >
          1
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <div
          aria-label="2nd February (Sunday)"
          class="rdp-day"
        >
          2
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="3rd February (Monday)"
          class="rdp-day"
        >
          3
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="4th February (Tuesday)"
          class="rdp-day"
        >
          4
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="5th February (Wednesday)"
          class="rdp-day"
        >
          5
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="6th February (Thursday)"
          class="rdp-day"
        >
          6
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="7th February (Friday)"
          class="rdp-day"
        >
          7
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="8th February (Saturday)"
          class="rdp-day"
        >
          8
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <div
          aria-label="9th February (Sunday)"
          class="rdp-day"
        >
          9
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="10th February (Monday)"
          class="rdp-day"
        >
          10
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="11th February (Tuesday)"
          class="rdp-day"
        >
          11
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="12th February (Wednesday)"
          class="rdp-day"
        >
          12
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="13th February (Thursday)"
          class="rdp-day"
        >
          13
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="14th February (Friday)"
          class="rdp-day"
        >
          14
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="15th February (Saturday)"
          class="rdp-day"
        >
          15
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <div
          aria-label="16th February (Sunday)"
          class="rdp-day"
        >
          16
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="17th February (Monday)"
          class="rdp-day"
        >
          17
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="18th February (Tuesday)"
          class="rdp-day"
        >
          18
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="19th February (Wednesday)"
          class="rdp-day"
        >
          19
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="20th February (Thursday)"
          class="rdp-day"
        >
          20
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="21st February (Friday)"
          class="rdp-day"
        >
          21
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="22nd February (Saturday)"
          class="rdp-day"
        >
          22
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <div
          aria-label="23rd February (Sunday)"
          class="rdp-day"
        >
          23
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="24th February (Monday)"
          class="rdp-day"
        >
          24
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="25th February (Tuesday)"
          class="rdp-day"
        >
          25
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="26th February (Wednesday)"
          class="rdp-day"
        >
          26
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="27th February (Thursday)"
          class="rdp-day"
        >
          27
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="28th February (Friday)"
          class="rdp-day"
        >
          28
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="29th February (Saturday)"
          class="rdp-day"
        >
          29
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`when showing the week numbers should render correctly 1`] = `
<table
  class="rdp-table"
  role="grid"
>
  <thead
    class="rdp-head"
  >
    <tr
      class="rdp-head_row"
    >
      <th
        class="rdp-head_cell"
        scope="col"
      />
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Su
        </span>
        <span
          class="rdp-vhidden"
        >
          Sunday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Mo
        </span>
        <span
          class="rdp-vhidden"
        >
          Monday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Tu
        </span>
        <span
          class="rdp-vhidden"
        >
          Tuesday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          We
        </span>
        <span
          class="rdp-vhidden"
        >
          Wednesday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Th
        </span>
        <span
          class="rdp-vhidden"
        >
          Thursday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Fr
        </span>
        <span
          class="rdp-vhidden"
        >
          Friday
        </span>
      </th>
      <th
        class="rdp-head_cell"
        scope="col"
      >
        <span
          aria-hidden="true"
        >
          Sa
        </span>
        <span
          class="rdp-vhidden"
        >
          Saturday
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    class="rdp-tbody"
  >
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <span
          class="rdp-weeknumber"
        >
          5
        </span>
      </td>
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      />
      <td
        class="rdp-cell"
      >
        <div
          aria-label="1st February (Saturday)"
          class="rdp-day"
        >
          1
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <span
          class="rdp-weeknumber"
        >
          6
        </span>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="2nd February (Sunday)"
          class="rdp-day"
        >
          2
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="3rd February (Monday)"
          class="rdp-day"
        >
          3
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="4th February (Tuesday)"
          class="rdp-day"
        >
          4
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="5th February (Wednesday)"
          class="rdp-day"
        >
          5
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="6th February (Thursday)"
          class="rdp-day"
        >
          6
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="7th February (Friday)"
          class="rdp-day"
        >
          7
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="8th February (Saturday)"
          class="rdp-day"
        >
          8
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <span
          class="rdp-weeknumber"
        >
          7
        </span>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="9th February (Sunday)"
          class="rdp-day"
        >
          9
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="10th February (Monday)"
          class="rdp-day"
        >
          10
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="11th February (Tuesday)"
          class="rdp-day"
        >
          11
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="12th February (Wednesday)"
          class="rdp-day"
        >
          12
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="13th February (Thursday)"
          class="rdp-day"
        >
          13
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="14th February (Friday)"
          class="rdp-day"
        >
          14
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="15th February (Saturday)"
          class="rdp-day"
        >
          15
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <span
          class="rdp-weeknumber"
        >
          8
        </span>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="16th February (Sunday)"
          class="rdp-day"
        >
          16
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="17th February (Monday)"
          class="rdp-day"
        >
          17
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="18th February (Tuesday)"
          class="rdp-day"
        >
          18
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="19th February (Wednesday)"
          class="rdp-day"
        >
          19
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="20th February (Thursday)"
          class="rdp-day"
        >
          20
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="21st February (Friday)"
          class="rdp-day"
        >
          21
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="22nd February (Saturday)"
          class="rdp-day"
        >
          22
        </div>
      </td>
    </tr>
    <tr
      class="rdp-row"
    >
      <td
        class="rdp-cell"
      >
        <span
          class="rdp-weeknumber"
        >
          9
        </span>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="23rd February (Sunday)"
          class="rdp-day"
        >
          23
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="24th February (Monday)"
          class="rdp-day"
        >
          24
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="25th February (Tuesday)"
          class="rdp-day"
        >
          25
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="26th February (Wednesday)"
          class="rdp-day"
        >
          26
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="27th February (Thursday)"
          class="rdp-day"
        >
          27
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="28th February (Friday)"
          class="rdp-day"
        >
          28
        </div>
      </td>
      <td
        class="rdp-cell"
      >
        <div
          aria-label="29th February (Saturday)"
          class="rdp-day"
        >
          29
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`when using custom components should render correctly 1`] = `
<table
  class="rdp-table"
  role="grid"
>
  <div>
    CustomHead
  </div>
  <tbody
    class="rdp-tbody"
  >
    <div>
      CustomRow
    </div>
    <div>
      CustomRow
    </div>
    <div>
      CustomRow
    </div>
    <div>
      CustomRow
    </div>
    <div>
      CustomRow
    </div>
  </tbody>
  <div>
    CustomFooter
  </div>
</table>
`;
